up-l1/zenoh.adoc

```diff
@@ -98,22 +108,21 @@ Different message types **MUST** use different Zenoh API.
 
 === URI Mapping
 
-UUri **MUST** be able to be serialized into link:../basics/uri.adoc#42-micro-uris[micro form].
-Then we can transform the micro-form UUri, which is `u8` array, into string in lowercase hexadecimal representation.
+UUri **MUST** be able to be serialized into link:../basics/uri.adoc#short-uris[short form].
+Then we can transform the short-form UUri into string for Zenoh key.
```

Comment:
Two questions:
- Does zenoh comply with the short form URI requirement that URIs are case insensitive? This matters for ID and IPv6 authorities - if zenoh is not case insensitive, then we need to add a requirement that case is normalized (e.g. lower-case the whole key)
- How are IP addresses normalized? Particularly with IPv6 (but also to a lesser extent IPv4), there are many ways to write the same address. If zenoh keys are a literal match, then we should probably define what the normalized address format is.

Reply:
We use the definition of short-form in up-spec
https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uri.adoc#51-remote-uris
However, you raise a good point. I'm not sure whether short form URI should take care of case sensitive or not.
@stevenhartley Do you know that?

Reply:
The case sensitivity of URIs are covered in the URI spec:

```
Schema and authority MUST be case-insensitive per RFC3986
Path, query, and message MUST be case-sensitive
```

If Zenoh key expressions cannot handle the case differences then I recommend we normalize it and define that in the zenoh.adoc spec.

Reply:
I thought the definition here was only for long-format before.
Ok, then we should normalize it to lowercase in Zenoh key.

Reply:
@evshary

I've looked at the change, and I'm not sure that it helps clarify the needs of normalization. The big concern for me is how IP addresses are normalized - for example, there are many ways to write an IPv6 address that all represent the same address but have different strings. It's not just whether the characters are upper case or lower case. Do we keep or drop leading zeros (both IPv6 and IPv4)? Always compact groups of zeros with `::` or always expand `::`? Do we convert the dual-format addresses (e.g. `2001:db8::f00f:192.168.1.4` or `::1.1.1.1`) to native IPv6 formats?

One option would be to parse the IP address string then convert it back to a string using the APIs provided by Linux, and that behavior becomes spec. Another would be to look at RFCs to see if there is a spec for normalization of addresses.

The second concern I have with the change is that the new language on converting to lower case is unclear. It sounds like the entire key would be converted to lower case, but _only the schema and authority_ should be case insensitive (and lower cased in the key), if I'm understanding what @stevenhartley said earlier..

Reply:
@gregmedd I think the best way forward is to explicitly state that the IP address must follow [RFC5952](https://www.rfc-editor.org/rfc/rfc5952). I added this to https://github.com/eclipse-uprotocol/up-spec/issues/115

Reply:
@gregmedd
For the first one, I think it will not be a problem if the IP address always follows RFC5952 in up-spec.
For the second one, yes, I should limit the transformation only to the schema and authority.

Reply:
Done.